shuup/admin/utils/product.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2021, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from django.http import HttpResponseRedirect
+from django.db.transaction import atomic
+from django.contrib import messages
+
+from shuup.core.models import (
+    Product, ProductAttribute, ProductMedia, ShopProduct
+)
+from shuup.admin.signals import product_copied
+from shuup.utils.models import get_data_dict
+
+class ProductCloner():
+    def __init__(self, current_shop, current_supplier):
+        self.current_shop = current_shop
+        self.current_supplier = current_supplier
+    
+    @atomic
+    def clone_product(self, shop_product=None):
+        product = shop_product.product
```

Comment:
maybe use `shuup.utils.models.copy_model_instance`, then update the SKU like you did and add a "Copy" suffix to the product name `new_product.name = _("{name} - Copy").format(name=product.name)`.
We also should go through all the translated fields and copy them (you need to find a way of getting the translated field dynamically, like [here in `_add_translated_columns`](shuup/admin/modules/settings/view_settings.py)).

Reply:
@nvthanh11 Sorry, here is the correct link: https://github.com/shuup/shuup/blob/master/shuup/admin/modules/settings/view_settings.py#L160